src/shared/components/list/list.component.tsx

```diff
@@ -1,55 +1,76 @@
-import React from 'react';
-import { FormattedMessage } from 'react-intl';
+import React, { useState } from 'react';
+import { sortBy, prop, toLower, compose } from 'ramda';
 
-import LineImg from '../../../images/line.png';
-import { highlightBlip, highlightLegend, unhighlightBlip } from '../../utils/radarUtils';
-import { color } from '../../../theme';
+import { useSelector } from 'react-redux';
 import {
-  Container,
-  SearchInputWrapper,
-  SearchWrapper,
-  SearchIcon,
-  Image,
-  ListWrapper,
-  SearchInput,
-  ListItem,
-} from './list.styles';
-import messages from './list.messages';
+  getBlipDataById,
+  hideBubble,
+  highlightBlip,
+  highlightLegend,
+  showBubble,
+  unhighlightBlip,
+} from '../../utils/radarUtils';
+import { color } from '../../../theme';
+import { RadarRing, RadarTechnology } from '../radar/radar.types';
+import { selectSearch } from '../../../modules/filters/filters.selectors';
+import { TagSize } from '../tag/tag.types';
+import { ListWrapper, ListItem, EmptyResults, ListLabel, ListItemTags, Tag } from './list.styles';
 
-// TODO List has to get same data as radar (with id,x,y)
-export const List = () => {
-  return (
-    <Container>
-      <SearchWrapper>
-        <SearchInputWrapper>
-          <FormattedMessage {...messages.searchPlaceholder}>
-            {(placeholder) => <SearchInput placeholder={placeholder.toString()} />}
-          </FormattedMessage>
-          <SearchIcon />
-        </SearchInputWrapper>
-        <Image src={LineImg} />
-      </SearchWrapper>
+interface ListProps {
+  technologies: RadarTechnology[];
+  emptyResults: boolean;
+  rings: RadarRing[];
+}
+
+export const List = ({ technologies, emptyResults, rings }: ListProps) => {
+  const [hoveredItem, setHoveredItem] = useState<string | null>(null);
+  const searchText = useSelector(selectSearch);
 
-      <ListWrapper>
-        {'123456789012345678901234'.split('').map((number, i) => (
-          <ListItem
-            key={i}
-            id={`list-item-${i}`}
-            onMouseEnter={() => {
-              // TODO pass real item data & show bubble
-              highlightBlip({ id: i.toString(), ring: 0 });
-              highlightLegend({ id: i.toString() });
-            }}
-            onMouseLeave={() => {
-              // TODO pass real item data & hide bubble
-              unhighlightBlip({ id: i.toString(), ring: 0, color: color.silver });
-              highlightLegend({ id: i.toString(), mode: 'off' });
-            }}
-          >
-            List item
-          </ListItem>
-        ))}
-      </ListWrapper>
-    </Container>
+  const sortedTechnologies = sortBy(compose(toLower, prop('label')), technologies);
+
+  if (emptyResults) {
+    return (
+      <EmptyResults>
+        We’re sorry, but we haven’t found any technology called “{searchText}”. Please try again and note that we don’t
```

Comment:
We should move it to `messages.ts`
and use `<FormattedMessage {messages.emptyString} values={{ searchText }} ...`